qa/common/common.sh

```diff
@@ -149,7 +167,39 @@ EOF
 #
 
 function proposal_populate_dmcrypt {
-    salt-run proposal.populate encryption='dmcrypt' name='dmcrypt'
+  salt-run proposal.populate encryption='dmcrypt' name='dmcrypt'
+}
+
+#
+# functions for restarting all configured services
+#
+
+function restart_services {
+  salt-run state.orch ceph.restart
+}
+
+function mon_restarted {
+  local exit_status 
+  local expectation=$1
+  salt --static --out json -C 'I@roles:mon' cmd.shell "journalctl -u ceph-mon@* " | grep -i terminated
+  exit_status={$?}
+  test ${exit_status} = ${expectation}
+}
+
+function osd_restarted {
+  local exit_status 
+  local expectation=$1
+  salt --static --out json -C 'I@roles:storage' cmd.shell "journalctl -u ceph-osd@* " | grep -i terminated
+  exit_status={$?}
+  test ${exit_status} = ${expectation}
+}
```

Comment:
Typically there are multiple `storage` nodes and each one has multiple `ceph-osd@*` services. For test to pass, it is enough if just one `ceph-osd@` unit restarts on one of the nodes?

Reply:
True; we could iterate over each OSD and invoke checks upon those.